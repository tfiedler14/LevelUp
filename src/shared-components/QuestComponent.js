import * as React from 'react';
import { Text, View } from 'react-native';
import { Col, Grid } from 'react-native-easy-grid';
import { Card } from 'react-native-paper';
import EStyleSheet from 'react-native-extended-stylesheet';
import { setLocation } from '../logic/location/actions';
import { connect } from 'react-redux';
import {deleteData, setQuest} from '../logic/data/actions';
import Icon from 'react-native-vector-icons/MaterialIcons';

export const QuestComponent = ({ info, setLocation, setQuest, deleteData }) => {
  return (
    <Card
      style={styles.card}
      onPress={() => {
        setQuest(info);
        setLocation('quest');
      }}>
      <Grid style={{ width: '100%' }}>
        <Col size={2}>
          <View style={styles.informationWrapper}>
              <Grid>
                <Col size={6}>
                  <View style={{ position: 'absolute' }}>
                    <Text style={styles.questName}>{info ? info.name : 'No Info'}</Text>
                  </View>
                </Col>

              </Grid>
              <View style={{ position: 'absolute' }}>
                <Text style={styles.questName}>{info ? info.name : 'No Info'}</Text>
              </View>
          </View>
          <View style={styles.informationWrapper}>
            <Text style={styles.description}>{info ? (info.description) : 'No Description'}</Text>

          </View>
        </Col>
      </Grid>
    </Card>
  );
};

const styles = EStyleSheet.create({
  card: {
    width: '100%',
    height: '5rem',
    //  border: 'none',
    marginBottom: '1rem',
    color: 'white',
    backgroundColor: '#555'
  },
  questnameContainer: {
    position: 'absolute',
    padding: '.5rem'
  },
  informationWrapper: {
    marginTop: '.5rem',
    marginBottom: '.5rem',
    marginRight: '.5rem',
    marginLeft: '.5rem'
  },

  questName: {
    fontWeight: 'bold',
    fontSize: '1rem',
    marginBottom: '1rem',
    color: 'white'
    // fontFamily: 'sans-serif',
  },

  description: {
    fontSize: '1rem',
    marginBottom: '.5rem',
    fontStyle: 'italic',
    color: 'white'
    // fontFamily: 'sans-serif',
  },

  subText: {
    fontSize: '.85rem',
    marginBottom: '.25rem'
    // fontFamily: 'sans-serif',
  }
});

const mapStateToProps = state => {
  return {
    location: state.location
  };
};
const mapDispatchToProps = dispatch => {
  return {
    setLocation: location => {
      dispatch(setLocation(location));
    },
    deleteData: (data, location) => {
      dispatch(deleteData(data, location));
    },
    setQuest: (data, location) => {
        dispatch(setQuest(data));
    }
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(QuestComponent);
