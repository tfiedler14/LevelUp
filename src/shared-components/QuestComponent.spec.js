import React from 'react';
import { shallow } from 'enzyme';
import { mapDispatchToProps, mapStateToProps, QuestComponent } from './QuestComponent';

describe('QuestComponent', () => {
  it('should render QuestComponent correctly', () => {
    const component = shallow(
      <QuestComponent
        setQuest={jest.fn()}
        deleteData={jest.fn()}
        setLocation={jest.fn()}
        info={{}}
      />
    );

    expect(component).toMatchSnapshot();
  });

  it('should render QuestComponent correctly when info is null', () => {
    const component = shallow(
      <QuestComponent
        setQuest={jest.fn()}
        deleteData={jest.fn()}
        setLocation={jest.fn()}
        info={null}
      />
    );

    expect(component).toMatchSnapshot();
  });
});

describe('mapStateToProps', () => {
  it('should return the correct auth, profile, and initialValues', () => {
    const state = {
      location: 'quest'
    };

    expect(mapStateToProps(state)).toEqual({
      location: 'quest'
    });
  });
});

describe('mapDispatchToProps', () => {
  it('should dispatch setQuest when setQuest is called', () => {
    const dispatch = jest.fn();
    const props = mapDispatchToProps(dispatch);
    props.setQuest({}, 'test');

    expect(dispatch).toHaveBeenCalledWith({ data: {}, type: 'SET_QUEST' });
  });

  it('should dispatch deleteData when deleteData is called', () => {
    const dispatch = jest.fn();
    const props = mapDispatchToProps(dispatch);
    props.deleteData({}, 'test');

    expect(dispatch).toHaveBeenCalledWith(expect.any(Function));
  });

  it('should dispatch setLocation when setLocation is called', () => {
    const dispatch = jest.fn();
    const props = mapDispatchToProps(dispatch);
    props.setLocation('home');

    expect(dispatch).toHaveBeenCalledWith({ location: 'home', type: 'SET_LOCATION' });
  });
});
