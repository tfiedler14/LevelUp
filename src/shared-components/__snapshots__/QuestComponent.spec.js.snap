// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestComponent should render QuestComponent correctly 1`] = `
<withTheme(Card)
  onPress={[Function]}
>
  <GridNB
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <ColumnNB
      size={2}
    >
      <View>
        <GridNB>
          <ColumnNB
            size={6}
          >
            <View
              style={
                Object {
                  "position": "absolute",
                }
              }
            >
              <Text />
            </View>
          </ColumnNB>
          <ColumnNB
            size={1}
          >
            <View
              style={
                Object {
                  "position": "absolute",
                }
              }
            >
              <Icon
                allowFontScaling={false}
                color="#999999"
                name="edit"
                onPress={[Function]}
                size={24}
              />
            </View>
          </ColumnNB>
          <ColumnNB
            size={1}
          >
            <View
              style={
                Object {
                  "position": "absolute",
                }
              }
            >
              <Icon
                allowFontScaling={false}
                color="#999999"
                name="delete"
                onPress={[Function]}
                size={24}
              />
            </View>
          </ColumnNB>
        </GridNB>
        <View
          style={
            Object {
              "position": "absolute",
            }
          }
        >
          <Text />
        </View>
      </View>
      <View>
        <Text>
          Description: undefined
        </Text>
      </View>
    </ColumnNB>
  </GridNB>
</withTheme(Card)>
`;

exports[`QuestComponent should render QuestComponent correctly when info is null 1`] = `
<withTheme(Card)
  onPress={[Function]}
>
  <GridNB
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <ColumnNB
      size={2}
    >
      <View>
        <GridNB>
          <ColumnNB
            size={6}
          >
            <View
              style={
                Object {
                  "position": "absolute",
                }
              }
            >
              <Text>
                No Info
              </Text>
            </View>
          </ColumnNB>
          <ColumnNB
            size={1}
          >
            <View
              style={
                Object {
                  "position": "absolute",
                }
              }
            >
              <Icon
                allowFontScaling={false}
                color="#999999"
                name="edit"
                onPress={[Function]}
                size={24}
              />
            </View>
          </ColumnNB>
          <ColumnNB
            size={1}
          >
            <View
              style={
                Object {
                  "position": "absolute",
                }
              }
            >
              <Icon
                allowFontScaling={false}
                color="#999999"
                name="delete"
                onPress={[Function]}
                size={24}
              />
            </View>
          </ColumnNB>
        </GridNB>
        <View
          style={
            Object {
              "position": "absolute",
            }
          }
        >
          <Text>
            No Info
          </Text>
        </View>
      </View>
      <View>
        <Text>
          No Description
        </Text>
      </View>
    </ColumnNB>
  </GridNB>
</withTheme(Card)>
`;
